clouddriver-ecs/src/test/groovy/com/netflix/spinnaker/clouddriver/ecs/deploy/EcsServerGroupNameResolverSpec.groovy

```diff
@@ -92,4 +110,85 @@ class EcsServerGroupNameResolverSpec extends Specification {
     nextServerGroupName == "application-stack-details-v003".toString()
     containerName == "v003"
   }
+
+  void "should skip names that already exist as ECS services in active and draining state"() {
+    given:
+    def resolver = new EcsServerGroupNameResolver(ecsClusterName, ecsClient, region)
+    ecsClient.listServices(_) >> new ListServicesResult().withServiceArns("application-stack-details-v001", "application-stack-details-v002")
+    ecsClient.describeServices({DescribeServicesRequest request ->
+      request.cluster == ecsClusterName
+      request.services == ["application-stack-details-v001", "application-stack-details-v002"]
+    }) >> new DescribeServicesResult().withServices(
+      new Service(serviceName: "application-stack-details-v001", createdAt: new Date(1), status: "ACTIVE"),
+      new Service(serviceName: "application-stack-details-v002", createdAt: new Date(2), status: "ACTIVE"))
+    ecsClient.describeServices({DescribeServicesRequest request ->
+      request.cluster == ecsClusterName
+      request.services == ["application-stack-details-v003"]
+    }) >> new DescribeServicesResult().withServices(
+      new Service(serviceName: "application-stack-details-v003", createdAt: new Date(3), status: "DRAINING"))
```

Comment:
what happens if the service returned is `INACTIVE` ? will we create a new server group with `-V003` ? this feels like weird behavior if there's an active service called `-V004` but i'm also not sure how to prevent it w/o describing until we find one that's `MISSING`

Reply:
Correct, v003 would be created.  The "should not skip names for inactive ECS services" unit test below covers this.

The resolver already has the full list of "taken slots" aka the list of active services, so it would know that there is a v004 and pick the right number appropriately.  This check just makes sure that we don't attempt to use a name that isn't a "taken slot", but can't actually be used.

Reply:
> The resolver already has the full list of "taken slots" aka the list of active services, so it would know that there is a v004 and pick the right number appropriately.

Ah ha! that's the piece i was missing. in that case :shipit: